### PR TITLE
adding method to calculate active case

### PR DESCRIPTION
**Pull request by osvyg**
Resolves https://github.com/lazd/coronadatascraper/issues/33

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -182,9 +182,21 @@ async function scrapeData() {
   return { locations };
 };
 
+function calculateActive(locations) {
+  return locations.map(l => {
```
Comment (lazd) [inline index.js]: Can we move this inner arrow function to `lib/transform.js` called `getActiveFromLocation`?

File: index.js [hunk 2]
```diff
@@ -182,9 +182,21 @@ async function scrapeData() {
...
+    let recovered = l["recovered"] !== undefined ? l["recovered"] : 0;
+    l["active"] = cases - deaths - recovered;
+    return l;
+  });
+}
+
 async function writeData({ locations, featureCollection }) {
   let date = process.env['SCRAPE_DATE'] ?  '-' + process.env['SCRAPE_DATE'] : '';
 
+  locations = calculateActive(locations);
```
Comment (lazd) [inline index.js]: Let's do this above inside of `scrape()`, that way it's available further down the pipeline and to the API.

Maybe you can add it to the loop that's currently calculating counties/states/countries so we're not looping twice.

File: index.js (hunk 1 above)
Comment (osvyg) [reply index.js]: fixed

File: index.js (hunk 2 above)
Comment (osvyg) [reply index.js]: fixed

**Comment by lazd** [comment]
Excellent work, thank you!

File: index.js [hunk 3]
```diff
@@ -124,6 +124,7 @@ async function scrapeData() {
     else {
       counties++;
     }
+    location["active"] = transform.getActiveFromLocation(location)
```
Comment (lazd) [inline index.js]: @osvyg just realized this should probably check if `location.active` is already defined... perhaps the scraper or datasource already has it... Should we use that value if so?

Comment (osvyg) [reply index.js]: True, ill do a check. I'll create a new PR

Comment (osvyg) [reply index.js]: Yeah, I think we should trust what the active data is coming back from the scraper. If none then we calculate it

**Pull request merged**